File: 14-LongestCommonPrefix/14-LongestCommonPrefix.cpp
// Last updated: 7/16/2026, 4:39:27 PM
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = "";
        sort(strs.begin(),strs.end());
        int n = strs.size();
        string first=strs[0],last=strs[n-1];
        for(int i=0;i<min(first.size(),last.size());i++){
            if(first[i]!=last[i]){
                return ans;
            }
            ans+=first[i];
        }
        return ans;
    }
};

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 8/26/2026, 10:04:14 AM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:

        strs.sort()
        first = strs[0]
        last = strs[-1]
        i = 0

        while i < len(first) and i < len(last) and first[i] == last[i]:
            i += 1
        return first[:i]

        